libtiny_tui/src/lib.rs

```diff
@@ -204,17 +202,17 @@ async fn input_handler<S>(
 
 macro_rules! delegate {
     ( $name:ident ( $( $x:ident: $t:ty, )* ) ) => {
-        fn $name(&self, $($x: $t,)*) {
+        pub fn $name(&self, $($x: $t,)*) {
             if let Some(inner) = self.inner.upgrade() {
                 inner.borrow_mut().$name( $( $x, )* );
             }
         }
     }
 }
 
-impl UI for TUI {
+impl TUI {
     delegate!(draw());
-    delegate!(new_server_tab(serv_name: &str, alias: Option<String>,));
+    delegate!(new_server_tab(serv_name: &str, alias: &Option<String>,));
```

Comment:
Why did you have to change this?

Reply:
To make this macro happy:
`ui.rs`
```
macro_rules! delegate {
    ( $name:ident ( $( $x:ident: $t:ty, )* )) => {
        pub(crate) fn $name(&self, $($x: $t,)*) {
            self.ui.$name( $( $x, )* );
            if let Some(logger) = &self.logger {
                logger.$name( $( $x, )* );
            }
        }
    }
}
```

Let me know if you can spot a better way. I'm a macro noob

Reply:
No need to change the macro, I added another comment below.

Reply:
@osa1 This is the reason why I changed it:
![image](https://user-images.githubusercontent.com/5448324/103022207-955add80-455c-11eb-9656-38f76b7c10f6.png)